dips/dip-182.md

```diff
@@ -0,0 +1,72 @@
+---
+dip: 182
+title: On-Chain Domain Resolution
+authors: Sunmi Lee (@sunmilee), David Wolinsky (@davidiw)
+status: Draft
+type: Informational
+created: 06/08/2021
+last_updated: 06/015/2021
+issue: https://github.com/diem/dip/issues/182
+---
+
+# Summary
+
+This DIP describes the on-chain domain name resolution process.
+
+On-chain domain name resolution allows organizations to be able to define their product names, and for VASPs to look up other VASPs using the product names.
+Having a human-readable VASP identifier gives the Diem Ecosystem the ability numerate all VASPs by their names, which is useful for merchant purposes as well as p2p transaction.
+For example, given a VASP domain name in a payment transaction, VASPs can look up where to route the payment correctly using an approved and accurate mapping of VASP identifier to an on-chain address.
+The Diem blockchain is a natural place to store VASP domain information for domain name resolution. We define here an on-chain domain resolution system that will allow VASPs to query for a VASP's address given a VASP domain.
+
+# On-chain data
+
+## Domain Format
+A VASP domain name is a unique string that is mapped to a VASP. Specification:
+* Case insensitive
+* Valid regular expression: `^[a-zA-Z0-9][a-zA-Z0-9.]*$`
+* Maximum length: 63 characters (64 including `@`)
+
+## DiemID Domain
+
+```
+resource struct DiemIdDomains {
+    domains: vector<DiemIdDomain>,
+}
+struct DiemIdDomain {
+    domain: vector<u8>,  // Utf-8 encoded
+}
+```
+* Field definitions:
+    * `domain` - name of a DiemID domain
+* The `DiemIdDomains` resource can only be published into an account with `Role == PARENT_VASP_ROLE_ID`.
```

Comment:
might be worth relabeling these if we push off on the 1.3 release.

Reply:
When is the next cut?

Reply:
1.3 or 1.4 won't matter since the framework release won't happen until 1.4 cut.